assets/js/src/index.js

```diff
@@ -30,3 +56,21 @@ function warnIfDataMissing() {
 		);
 	}
 }
+
+document.addEventListener( 'wp_listen_for_consent_change', function ( event ) {
+	const consentUpdate = {};
+
+	const types = consentMap[ Object.keys( event.detail )[ 0 ] ];
+	const state =
+		Object.values( event.detail )[ 0 ] === 'allow' ? 'granted' : 'deny';
+
+	if ( types !== undefined ) {
+		types.forEach( ( type ) => {
+			consentUpdate[ type ] = state;
+		} );
+
+		if ( Object.keys( consentUpdate ).length > 0 ) {
+			gtag( 'consent', 'update', consentUpdate ); // eslint-disable-line no-undef
```

Comment:
:construction: 
The `gtag` may not be set . It could be called differently, as per https://github.com/woocommerce/woocommerce-google-analytics-integration/blob/4c3a0362f33b32f755884b0c03c338f7a1f71b1a/includes/class-wc-google-gtag-js.php#L94:L94

```suggestion
			const gtag = window[ ga4w.settings.tracker_function_name ]; // eslint-disable-line camelcase -- This is a PHP-set config.
			gtag( 'consent', 'update', consentState );
```